zipp/compat/py313.py

```diff
@@ -0,0 +1,34 @@
+import functools
+import sys
+
+
+# from jaraco.functools 4.1
+def identity(x):  # pragma: no cover
+    return x
+
+
+# from jaraco.functools 4.1
+def apply(transform):  # pragma: no cover
+    def wrap(func):
+        return functools.wraps(func)(compose(transform, func))
+
+    return wrap
+
+
+# from jaraco.functools 4.1
+def compose(*funcs):  # pragma: no cover
+    def compose_two(f1, f2):
+        return lambda *args, **kwargs: f1(f2(*args, **kwargs))
+
+    return functools.reduce(compose_two, funcs)
+
+
+def replace(pattern):
+    r"""
+    >>> replace(r'foo\z')
+    'foo\\Z'
+    """
+    return pattern[:-2] + pattern[-2:].replace(r'\z', r'\Z')
```

Comment:
This could be made more generic, such as supporting `\z` not as the last pair of characters and supporting `\\z` (not replacing it), but I'm going to ignore those use-cases until they're potentially needed (such as making this compatibility shim more widely available).